workalendar/usa/core.py

```diff
@@ -259,6 +261,15 @@ def get_national_memorial_day(self, year):
             self.national_memorial_day_label
         )
 
+    def get_juneteenth_day(self, year):
+        """
+        Return Juneteenth Day
+        """
+        # Juneteenth started to be a federal holiday in 2021
+        if year < 2021:
+            raise ValueError("Juneteenth became a federal holiday in 2021")
+        return (date(year, 6, 19), "Juneteenth National Independence Day")
```

Comment:
(just for the sake of it, I double checked the official holiday name: https://en.wikipedia.org/wiki/Juneteenth :heavy_check_mark: )